src/utils/SessionLock.ts

```diff
@@ -242,13 +245,25 @@ export async function getSessionLock(onNewInstance: () => Promise<void>): Promis
             };
         });
 
+        let sleepExpired = false;
         const sleepPromise = new Promise((resolve) => {
-            setTimeout(resolve, remaining, undefined);
+            setTimeout(() => {
+                sleepExpired = true;
+                resolve(undefined);
+            }, remaining);
         });
```

Comment:
We also have a `sleep` util for this pattern already

Reply:
Yes, but it doesn't play nicely with the slightly creative tricks we play in the tests to test two instances of the lock together.